locations/storefinders/storerocket.py

```diff
@@ -41,3 +44,25 @@ def parse(self, response, **kwargs):
 
     def parse_item(self, item: Feature, location: dict, **kwargs):
         yield item
+
+    def storefinder_exists(response: Response) -> bool | Request:
+        # Example: # https://barre3.com/studio-locations
+        # Only after DOM load
+        # <script id="store-rocket-script" src="https://cdn.storerocket.io/widget.js"></script>
+
+        # Example: https://soapynoble.com/
+        # <script type="text/javascript" async="" src="https://cdn.storerocket.io/js/widget-mb.js"></script>
+        if response.xpath('//script[contains(@src, "https://cdn.storerocket.io/widget")]').get():
+            return True
+
+        # Example: https://soapynoble.com/
+        # Example: https://ribcrib.com/locations/
+        if response.xpath('//div[@id="storerocket-widget"]').get():
+            return True
+
+        return False
+
+    def extract_spider_attributes(response: Response) -> dict | Request:
+        return {
+            "allowed_domains": [urlparse(response.url).netloc],
```

Comment:
We should be returning `storerocket_id` here (or rename this attribute to `api_key`).

For example https://soapynoble.com/
`<div id='storerocket-widget' style='width:100%;' data-storerocket-env='p' data-storerocket-id='vo8xZwy4gn'>`

For example https://ribcrib.com/locations/
`<div id='storerocket-widget' style='width:100%;' data-storerocket-env='p' data-storerocket-id='6wgpr528XB'>`